sklearn/utils/estimator_checks.py

```diff
@@ -1854,26 +1837,18 @@ def check_classifiers_one_label(name, classifier_orig, strict_mode=True):
     # catch deprecation warnings
     with ignore_warnings(category=FutureWarning):
         classifier = clone(classifier_orig)
-        # try to fit
-        try:
+        with raises(
+            ValueError, match="class", may_pass=True, err_msg=error_string_fit
+        ) as cm:
             classifier.fit(X_train, y)
-        except ValueError as e:
-            if 'class' not in repr(e):
-                print(error_string_fit, classifier, e)
-                traceback.print_exc(file=sys.stdout)
-                raise e
-            else:
-                return
-        except Exception as exc:
-            print(error_string_fit, classifier, exc)
-            traceback.print_exc(file=sys.stdout)
-            raise exc
-        # predict
-        try:
-            assert_array_equal(classifier.predict(X_test), y)
-        except Exception as exc:
-            print(error_string_predict, classifier, exc)
-            raise exc
+
+        if cm.raised_and_matched:
+            # ValueError was raised with proper error message
+            return
+
+        assert_array_equal(
+            classifier.predict(X_test), y, err_msg=error_string_predict
+        )
```

Comment:
This is the only place where we use `raised_and_matched`.

Do we have any classifier that actually raises an error with "Classifier can't predict predict when only one class is present"?

Reply:
I think so, because otherwise CodeCov would give us a warning here on the GitHub GUI.

Also note that the error isn't expected in predict but in `fit`.